app/src/main/java/org/fossasia/openevent/app/contract/view/LoginView.java

```diff
@@ -0,0 +1,11 @@
+package org.fossasia.openevent.app.contract.view;
+
+public interface LoginView {
+
+    void showProgressBar(boolean show);
```

Comment:
View should be dumb, not knowing what the logic of doing anything should be. What you are saying is the MVC approach, we don't want that. We want presenter to decide what to do with view, view should just do what it is told. showData means show data, not stop loading, save in cache, update model. Everything requiring condition and login must be done in model or presenter

The onError and onSuccess are too generic, it should rather be, showError and showEventActivity because view should not know what success or error mean. Besides if we let view decide that, how we test that view actually calls it? That was the sole reason we were moving logic away from view to abstractions

Also, showError doesn't necessarily mean the loading was on. Suppose that we verify email format before sending it to server, we will still show error but not show the progressbar, this should thus be handled by presenter